Group 5 Final Project Submission

m-d-jo opened the pull request:
We are proposing a two-player networked version of the board game Quoridor.

Commit:
Create proposal.md

Commit:
Spelling and added Node/Express

charlieroberts commented:
This seems like a great project idea, and I think it's pretty doable. Socket.io (or vanilla WebSockets) could take the place of a lot of the middleware you're thinking about, but I think you could also pull this off with regular get/post requests. But I'd say that since some of you are IMGD folks, learning WebSockets/Socket.io makes a lot of sense for doing this type of multiplayer game in general.

Looking forward to playing this! I wasn't familiar with Quoridor but it sounds like the type of game I enjoy.

Commit:
Setup files + basic websocket functions

Commit:
Changed websockets -> ws

Commit:
Started on board helper functions

rjstebe committed:
finished IsValidPawnMove

rjstebe committed:
basic wall placement check

rjstebe committed:
Room object and joining rooms

Commit:
Pathfinding

Added pathfinding (simple depth-first), ASCII board printer, sample game state, board helper bug fixes

Commit:
Update log

Commit:
Update README.md

Commit:
Text client?

rjstebe committed:
finished server logic

rjstebe committed:
Merge branch 'server-progress' of https://github.com/m-d-jo/final_project into server-progress

rjstebe committed:
fixed room full send call

rjstebe committed:
Merge pull request #1 from m-d-jo/server-progress

Server progress

markjforte2000 committed:
Pawns able to move

Commit:
Fix input bug

Commit:
Merge branch 'main' of https://github.com/m-d-jo/final_project

rjstebe committed:
delete room when last player disconnects

rjstebe committed:
Merge pull request #2 from m-d-jo/server-progress

delete room when last player disconnects

rjstebe committed:
prevent player from passing turn by not moving

rjstebe committed:
fixed player 1 reconnecting to a room crashing server

Commit:
Fix bug preventing corner jumps

Commit:
Merge branch 'main' of https://github.com/m-d-jo/final_project

palomagonzalez committed:
Placing walls functionality

rjstebe committed:
added express file server to server.js

rjstebe committed:
Merge branch 'main' of https://github.com/m-d-jo/final_project

rjstebe committed:
removed board.js script tag in header

palomagonzalez committed:
Styling changes

palomagonzalez committed:
Merge branch 'main' of https://github.com/m-d-jo/final_project into main

Commit:
Add pathfinding check to wall placement

Commit:
Merge branch 'main' of https://github.com/m-d-jo/final_project

rjstebe committed:
server now doesn't validate moves after game has ended

rjstebe committed:
Merge branch 'main' of https://github.com/m-d-jo/final_project

markjforte2000 committed:
UI addons

markjforte2000 committed:
Merge remote-tracking branch 'origin/main'

rjstebe committed:
fix gameOver is not a function error

rjstebe committed:
Merge branch 'main' of https://github.com/m-d-jo/final_project

Commit:
Fix pathfinding bug pt. 2

Commit:
Merge branch 'main' of https://github.com/m-d-jo/final_project

Commit:
Reset invalid test walls before returning

rjstebe committed:
extended check for no room code

rjstebe committed:
Merge branch 'main' of https://github.com/m-d-jo/final_project

rjstebe committed:
send room code to server under correct key name

rjstebe committed:
fix syntax for adding room key to message data

markjforte2000 committed:
UI updates room code

Commit:
Print JSON string instead of object

Commit:
Merge branch 'main' of https://github.com/m-d-jo/final_project

markjforte2000 committed:
Clicking friend links opens new tab

markjforte2000 committed:
Merge remote-tracking branch 'origin/main'

rjstebe committed:
swap port numbers

rjstebe committed:
Merge branch 'main' of https://github.com/m-d-jo/final_project

markjforte2000 committed:
working for glitch

markjforte2000 committed:
Merge remote-tracking branch 'origin/main'

# Conflicts:
#	server.js

Commit:
Update README.md

Commit:
Remove client logs, general cleanup

m-d-jo committed:
Added video link
